micronota/commands/database.py

```diff
@@ -6,22 +6,46 @@
 # The full license is in the file COPYING.txt, distributed with this software.
 # ----------------------------------------------------------------------------
 
+import os
+import pkgutil
+import importlib
+
 import click
 
 from ..cli import cmd, AliasedGroup
+from .. import db
 
 
 @cmd.group(cls=AliasedGroup)
 @click.pass_context
 def cli(ctx):
     '''Database operations.'''
+    print(ctx.parent.params)
     pass
 
 
-@cli.command('update')
-@click.option('-i', '--input', type=click.STRING)
-@click.option('-o', '--output', type=click.STRING)
+@cli.command('prepare')
+@click.argument('databases', nargs=-1)
+@click.option('-f', '--force', is_flag=True,
+              help='Force overwrite.')
 @click.pass_context
-def update_db():
-    '''Update database.'''
-    pass
+def create_db(ctx, databases, force):
+    '''Prepare database.'''
```

Comment:
it is command line api and documented at click.option/argument

On Sun, Jan 31, 2016 at 1:05 PM, mortonjt <notifications@github.com> wrote:

> In micronota/commands/database.py
> <https://github.com/biocore/micronota/pull/40#discussion_r51367386>:
>
> >  @click.pass_context
> > -def update_db():
> > -    '''Update database.'''
> > -    pass
> > +def create_db(ctx, databases, force):
> > +    '''Prepare database.'''
>
> Is this a public facing function? Do we need complete doc-strings here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/micronota/pull/40/files#r51367386>.
>